[BUG] V3.0.0#beta PDF-Editor split not functional

### Description

Hi, successfully created a clone of my running instance and updated to v3.0.0#beta
Very excited started to check.
When trying to split a pdf ith the PDF-Editor i run into an error:

When trying to split a 5 pager to three separate documents. Setting: new document, copy metadata, delete original.

Any advise?

Best 
Sven

### Steps to reproduce

open a document
actions
pdf-editor
split marker (scissors)
settings: new file,copy metadata, delete original
OK

### Webserver logs

```bash
[2026-05-24 09:31:24,534] [INFO] [paperless.bulk_edit] Editing PDF of document 45993 with 3 operations

[2026-05-24 09:31:24,668] [INFO] [paperless.bulk_edit] Released archive serial numbers for documents []

[2026-05-24 09:31:24,699] [INFO] [paperless.bulk_edit] Restored archive serial numbers for documents []

[2026-05-24 09:31:24,699] [ERROR] [paperless.bulk_edit] Error editing document 45993: Cannot add link to group: do that on individual tasks

Traceback (most recent call last):

  File "/usr/src/paperless/src/documents/bulk_edit.py", line 859, in edit_pdf

    ).apply_async(

      ^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/site-packages/celery/canvas.py", line 2156, in apply_async

    return self.run(tasks, body, args, task_id=task_id, kwargs=kwargs, **merged_options)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/site-packages/celery/canvas.py", line 2256, in run

    header_result = header.apply_async(partial_args, kwargs, task_id=group_id, **options)

                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/site-packages/celery/canvas.py", line 1601, in apply_async

    raise TypeError(

TypeError: Cannot add link to group: do that on individual tasks

[2026-05-24 09:31:24,701] [WARNING] [paperless.api] An error occurred performing PDF edit: An error occurred while editing the document: Cannot add link to group: do that on individual tasks
```

### Browser logs

```bash

```

### Services logs

```bash

```

### Paperless-ngx version

3.0.0

### Host OS

Ubuntu 24.04

### Installation method

Docker - official image

### System status

```json

```

### Browser

Firefox

### Configuration changes

_No response_

### Please confirm the following

- [x] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [x] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [x] I have already searched for relevant existing issues and discussions before opening this report.
- [x] I have updated the title field above with a concise description.

--- a/src/documents/bulk_edit.py
+++ b/src/documents/bulk_edit.py
@@ -676,9 +676,9 @@ def split(
                     chord(
                         header=consume_tasks,
                         body=delete.si([doc.id]),
-                    ).apply_async(
-                        link_error=[restore_archive_serial_numbers_task.s(backup)],
-                    )
+                    ).on_error(
+                        restore_archive_serial_numbers_task.s(backup),
+                    ).apply_async()
                 except Exception:
                     restore_archive_serial_numbers(backup)
                     raise
@@ -856,9 +856,9 @@ def edit_pdf(
                     chord(
                         header=consume_tasks,
                         body=delete.si([doc.id]),
-                    ).apply_async(
-                        link_error=[restore_archive_serial_numbers_task.s(backup)],
-                    )
+                    ).on_error(
+                        restore_archive_serial_numbers_task.s(backup),
+                    ).apply_async()
                 except Exception:
                     restore_archive_serial_numbers(backup)
                     raise

--- a/src/documents/tests/test_bulk_edit.py
+++ b/src/documents/tests/test_bulk_edit.py
@@ -945,6 +945,10 @@ class TestPDFActions(DirectoriesMixin, TestCase):
         pages = [[1, 2], [3]]
         self.doc2.archive_serial_number = 200
         self.doc2.save()
+        errback = bulk_edit.restore_archive_serial_numbers_task.s(
+            {self.doc2.id: 200},
+        )
+        mock_chord.return_value.on_error.return_value = mock_chord.return_value
 
         result = bulk_edit.split(doc_ids, pages, delete_originals=True)
         self.assertEqual(result, "OK")
@@ -957,6 +961,8 @@ class TestPDFActions(DirectoriesMixin, TestCase):
 
         mock_delete_documents.assert_called()
         mock_chord.assert_called_once()
+        mock_chord.return_value.on_error.assert_called_once_with(errback)
+        mock_chord.return_value.apply_async.assert_called_once_with()
 
         delete_documents_args, _ = mock_delete_documents.call_args
         self.assertEqual(
@@ -991,6 +997,7 @@ class TestPDFActions(DirectoriesMixin, TestCase):
         self.doc2.save()
 
         sig = mock.Mock()
+        sig.on_error.return_value = sig
         sig.apply_async.side_effect = Exception("boom")
         mock_chord.return_value = sig
 
@@ -1256,10 +1263,16 @@ class TestPDFActions(DirectoriesMixin, TestCase):
         operations = [{"page": 1}, {"page": 2}]
         self.doc2.archive_serial_number = 250
         self.doc2.save()
+        errback = bulk_edit.restore_archive_serial_numbers_task.s(
+            {self.doc2.id: 250},
+        )
+        mock_chord.return_value.on_error.return_value = mock_chord.return_value
 
         result = bulk_edit.edit_pdf(doc_ids, operations, delete_original=True)
         self.assertEqual(result, "OK")
         mock_chord.assert_called_once()
+        mock_chord.return_value.on_error.assert_called_once_with(errback)
+        mock_chord.return_value.apply_async.assert_called_once_with()
         self.assertEqual(mock_consume_file.call_args.kwargs["overrides"].asn, 250)
         self.doc2.refresh_from_db()
         self.assertIsNone(self.doc2.archive_serial_number)
@@ -1288,6 +1301,7 @@ class TestPDFActions(DirectoriesMixin, TestCase):
         self.doc2.save()
 
         sig = mock.Mock()
+        sig.on_error.return_value = sig
         sig.apply_async.side_effect = Exception("boom")
         mock_chord.return_value = sig
 

